# Assignment #2: Data Exploration

In this assignment, you will demonstrate your knowledge of the Python and Pandas skills we've learned so far. These include:

- Getting an overview of your data
- Extracting columns
- Removing duplicates from your data
- Creating a subset of your data by matching a string
- Getting a random sample of your data
- Sorting your data 
- Getting fundamental statistical information of a series, such as the mean and median

You will not need to formulate or answer a research question for this assignment. You simply need to demonstrate your ability to perform specific operations on the data set you have selected. In the next assignment, your midterm, you will create a narrative exploration of the data using many of the same methods, but in such a way as to use narrative along with data analysis to answer a research question.

You may use the data set you imported in assignment #1, or you may choose another data set. That means that you can reuse code you wrote in Assignment #1 to import the data.

In the below notebook, I will ask for you to perform a task. Each task will be followed by one or more blank code cells. Use as many cells as you need to complete the task. If you know some other way to complete the task with Python and/or Pandas that is not the way we learned in class, that is also acceptable. Feel free to use Google searches to refresh your memory as to how to perform each task. You will also likely wish to consult the class notebooks, which can be found in [our class repository on GitHub](https://github.com/sha256rma/foundations-of-data-science).

If your data set does not have a needed form of data for a task, write code that would perform the task if a column with that data existed, and provide an explanation as a comment or markdown cell.

You may receive partial credit for some incomplete or incorrect answers. Please feel free to add comments about your thought process, which will make it more likely that you will receive partial credit.

The last cell is a bonus. You will not be penalized for not attempting or completing it.

-----

Import Pandas and read your data in as a data frame. Assign the data frame to a variable, such as `df`.

In [18]:
import pandas 

In [20]:
df = pandas.read_csv('data.csv')

df


name  reign  days       date              event  \
0        Buddy Rogers      1    36  11-Apr-63         House show   
1    Bruno Sammartino      1  2803  17-May-63         House show   
2         Ivan Koloff      1    21  18-Jan-71         House show   
3       Pedro Morales      1  1027   8-Feb-71         House show   
4        Stan Stasiak      1     9   1-Dec-73         House show   
..                ...    ...   ...        ...                ...   
407       Becky Lynch      3    41   8-Apr-19    WrestleMania 35   
408   Charlotte Flair      4     0  19-May-19  Money in the Bank   
409            Bayley      1   140  19-May-19  Money in the Bank   
410   Charlotte Flair      5     5   6-Oct-19     Hell in a Cell   
411            Bayley      2   247  11-Oct-19          SmackDown   

                location                           link  \
0       Washington, D.C.  /wiki/Buddy_Rogers_(wrestler)   
1           New York, NY         /wiki/Bruno_Sammartino   
2           New York, NY              /wiki/Ivan_Koloff   
3           New York, NY            /wiki/Pedro_Morales   
4       Philadelphia, PA             /wiki/Stan_Stasiak   
..                   ...                            ...   
407  East Rutherford, NJ              /wiki/Becky_Lynch   
408         Hartford, CT          /wiki/Charlotte_Flair   
409         Hartford, CT        /wiki/Bayley_(wrestler)   
410       Sacramento, CA          /wiki/Charlotte_Flair   
411         Paradise, NV        /wiki/Bayley_(wrestler)   

                                   belt date_of_birth date_of_death  \
0                      WWE Championship     2/20/1921     6/26/1992   
1                      WWE Championship     10/6/1935     4/18/2018   
2                      WWE Championship     8/25/1942     2/18/2017   
3                      WWE Championship    10/22/1942     2/12/2019   
4                      WWE Championship     4/13/1937     6/19/1997   
..                                  ...           ...           ...   
407  WWE SmackDown Women's Championship     1/30/1987           NaN   
408  WWE SmackDown Women's Championship      4/5/1986           NaN   
409  WWE SmackDown Women's Championship     6/15/1989           NaN   
410  WWE SmackDown Women's Championship      4/5/1986           NaN   
411  WWE SmackDown Women's Championship     6/15/1989           NaN   

                                      cause_of_death  
0                                                NaN  
1    Multiple organ failure due to heart problems[2]  
2                                       Liver cancer  
3                                                NaN  
4                                      Heart failure  
..                                               ...  
407                                              NaN  
408                                              NaN  
409                                              NaN  
410                                              NaN  
411                                              NaN  

[412 rows x 11 columns]

In [30]:
df.loc[df['belt'] == "WWE United States Championship", ["name", "days"]][:10].sort_values(by=["days"], ascending = False)

name  days
227  Blackjack Mulligan   217
231  Blackjack Mulligan   204
223         Harley Race   183
226          Paul Jones   107
224    Johnny Valentine    93
228          Paul Jones    43
232         Bobo Brazil    22
225          Terry Funk    18
229  Blackjack Mulligan    11
230          Paul Jones     6

Make the data frame visible by placing the data frame variable in a cell by itself.

In [ ]:
df = pandas.read_csv('data.csv')

'df'


What columns are in your data frame? Use a function or method that shows all the columns.

In [46]:
wrestler_data = [['Bruno Sammartino', 1,	2803, '17-May-63', 'House show', 'New York, NY', '/wiki/Bruno_Sammartino', 'WWE Championship', '10/6/1935', '4/18/2018', 'Multiple organ failure due to heart problems[2]']]


In [48]:
type(wrestler_data)

list

In [50]:
columns = ['name', 'reign', 'days', 'date', 'event', 'location', 'link', 'belt', 'date_of_birth', 'date_of_death', 'cause_of_death']

In [52]:
type(columns)

list

In [54]:
wrestlers = pandas.DataFrame(data=wrestler_data, columns=columns)

In [56]:
wrestlers

name  reign  days       date       event      location  \
0  Bruno Sammartino      1  2803  17-May-63  House show  New York, NY   

                     link              belt date_of_birth date_of_death  \
0  /wiki/Bruno_Sammartino  WWE Championship     10/6/1935     4/18/2018   

                                    cause_of_death  
0  Multiple organ failure due to heart problems[2]

Write Python code to output the number of rows in your data frame.

In [102]:
rows = len(pandas.read_csv('data.csv'))

In [104]:
print(f"The wrestling dataset has {rows} rows")

The wrestling dataset has 412 rows


Write code to output a column from your data frame as a series. (Extract a column.)

In [115]:
df

name  reign  days       date              event  \
0        Buddy Rogers      1    36  11-Apr-63         House show   
1    Bruno Sammartino      1  2803  17-May-63         House show   
2         Ivan Koloff      1    21  18-Jan-71         House show   
3       Pedro Morales      1  1027   8-Feb-71         House show   
4        Stan Stasiak      1     9   1-Dec-73         House show   
..                ...    ...   ...        ...                ...   
407       Becky Lynch      3    41   8-Apr-19    WrestleMania 35   
408   Charlotte Flair      4     0  19-May-19  Money in the Bank   
409            Bayley      1   140  19-May-19  Money in the Bank   
410   Charlotte Flair      5     5   6-Oct-19     Hell in a Cell   
411            Bayley      2   247  11-Oct-19          SmackDown   

                location                           link  \
0       Washington, D.C.  /wiki/Buddy_Rogers_(wrestler)   
1           New York, NY         /wiki/Bruno_Sammartino   
2           New York, NY              /wiki/Ivan_Koloff   
3           New York, NY            /wiki/Pedro_Morales   
4       Philadelphia, PA             /wiki/Stan_Stasiak   
..                   ...                            ...   
407  East Rutherford, NJ              /wiki/Becky_Lynch   
408         Hartford, CT          /wiki/Charlotte_Flair   
409         Hartford, CT        /wiki/Bayley_(wrestler)   
410       Sacramento, CA          /wiki/Charlotte_Flair   
411         Paradise, NV        /wiki/Bayley_(wrestler)   

                                   belt date_of_birth date_of_death  \
0                      WWE Championship     2/20/1921     6/26/1992   
1                      WWE Championship     10/6/1935     4/18/2018   
2                      WWE Championship     8/25/1942     2/18/2017   
3                      WWE Championship    10/22/1942     2/12/2019   
4                      WWE Championship     4/13/1937     6/19/1997   
..                                  ...           ...           ...   
407  WWE SmackDown Women's Championship     1/30/1987           NaN   
408  WWE SmackDown Women's Championship      4/5/1986           NaN   
409  WWE SmackDown Women's Championship     6/15/1989           NaN   
410  WWE SmackDown Women's Championship      4/5/1986           NaN   
411  WWE SmackDown Women's Championship     6/15/1989           NaN   

                                      cause_of_death  
0                                                NaN  
1    Multiple organ failure due to heart problems[2]  
2                                       Liver cancer  
3                                                NaN  
4                                      Heart failure  
..                                               ...  
407                                              NaN  
408                                              NaN  
409                                              NaN  
410                                              NaN  
411                                              NaN  

[412 rows x 11 columns]

In [118]:
df['name']

0          Buddy Rogers
1      Bruno Sammartino
2           Ivan Koloff
3         Pedro Morales
4          Stan Stasiak
             ...       
407         Becky Lynch
408     Charlotte Flair
409              Bayley
410     Charlotte Flair
411              Bayley
Name: name, Length: 412, dtype: object

### Read the example below to **count** duplicates based on specific columns (e.g., 'Name' and 'Age')

Check duplicates based on 'Name' and 'Age' columns (modify the subset for your own columns):

`duplicates_subset = df.duplicated(subset=['Name', 'Age'])`

Count the number of duplicates in the subset:

`num_duplicates_subset = duplicates_subset.sum()`

Display no. of duplicate rows based on your column(s):

`print(f"\nNumber of duplicate rows based on 'Name' and 'Age': {num_duplicates_subset}")`


After reading the above, modify the code snippets to check for duplicate columns in your own data frame

In [179]:
duplicates_subset = df.duplicated(subset=['name', 'reign', 'days', 'date', 'event', 'location', 'link', 'belt', 'date_of_birth', 'date_of_death', 'cause_of_death'])

In [181]:
num_duplicates_subset = duplicates_subset.sum()

In [191]:
print(f"\nNumber of duplicate rows based on 'name', 'reign', 'days', 'date', 'event', 'location', 'link', 'belt', 'date_of_birth', 'date_of_death', and 'cause_of_death'] : {num_duplicates_subset}")


Number of duplicate rows based on 'name', 'reign', 'days', 'date', 'event', 'location', 'link', 'belt', 'date_of_birth', 'date_of_death', and 'cause_of_death'] : 0


If any duplicates exist within a column, remove them with the `.drop_duplicates()` function

In [194]:
df['name']

0          Buddy Rogers
1      Bruno Sammartino
2           Ivan Koloff
3         Pedro Morales
4          Stan Stasiak
             ...       
407         Becky Lynch
408     Charlotte Flair
409              Bayley
410     Charlotte Flair
411              Bayley
Name: name, Length: 412, dtype: object

Write Python code to output the number of rows in your data frame.

In [201]:
df.drop_duplicates(subset='name', inplace=True)

In [203]:
df

name  reign  days       date                          event  \
0        Buddy Rogers      1    36  11-Apr-63                     House show   
1    Bruno Sammartino      1  2803  17-May-63                     House show   
2         Ivan Koloff      1    21  18-Jan-71                     House show   
3       Pedro Morales      1  1027   8-Feb-71                     House show   
4        Stan Stasiak      1     9   1-Dec-73                     House show   
..                ...    ...   ...        ...                            ...   
394      Apollo Crews      1    20  25-May-20                            Raw   
397             Naomi      1     9  12-Feb-17            Elimination Chamber   
400           Natalya      1    86  20-Aug-17                     SummerSlam   
402          Carmella      1   131  10-Apr-18                      SmackDown   
405             Asuka      1   100  16-Dec-18  TLC: Tables, Ladders & Chairs   

             location                           link  \
0    Washington, D.C.  /wiki/Buddy_Rogers_(wrestler)   
1        New York, NY         /wiki/Bruno_Sammartino   
2        New York, NY              /wiki/Ivan_Koloff   
3        New York, NY            /wiki/Pedro_Morales   
4    Philadelphia, PA             /wiki/Stan_Stasiak   
..                ...                            ...   
394       Orlando, FL             /wiki/Apollo_Crews   
397       Phoenix, AZ         /wiki/Naomi_(wrestler)   
400      Brooklyn, NY         /wiki/Natalya_Neidhart   
402   New Orleans, LA      /wiki/Carmella_(wrestler)   
405      San Jose, CA         /wiki/Asuka_(wrestler)   

                                   belt date_of_birth date_of_death  \
0                      WWE Championship     2/20/1921     6/26/1992   
1                      WWE Championship     10/6/1935     4/18/2018   
2                      WWE Championship     8/25/1942     2/18/2017   
3                      WWE Championship    10/22/1942     2/12/2019   
4                      WWE Championship     4/13/1937     6/19/1997   
..                                  ...           ...           ...   
394      WWE United States Championship     8/22/1987           NaN   
397  WWE SmackDown Women's Championship    11/30/1987           NaN   
400  WWE SmackDown Women's Championship     5/27/1982           NaN   
402  WWE SmackDown Women's Championship    10/23/1987           NaN   
405  WWE SmackDown Women's Championship     9/26/1981           NaN   

                                      cause_of_death  
0                                                NaN  
1    Multiple organ failure due to heart problems[2]  
2                                       Liver cancer  
3                                                NaN  
4                                      Heart failure  
..                                               ...  
394                                              NaN  
397                                              NaN  
400                                              NaN  
402                                              NaN  
405                                              NaN  

[175 rows x 11 columns]

Compare the new value with the first time you outputted the number of rows above. Is it more, less, or the same?

In this scenario, the number of rows decreased drastically. The original unfiltered data set contained 412 rows x 11 columns, while the filtered data set retrieved 175 rows x 11 columns. 237 rows less.

Create a subset of the data that matches a specific string in a column. That is, extract all rows of the original data frame that contain a specific string in one of the columns. Save the resulting data frame to a variable. (Add more cells if needed)

In [24]:
filtering_condition = df['event'] == 'House show'
df_subset = df.loc[filtering_condition]

In [28]:
df_subset

name  reign  days       date       event  \
0              Buddy Rogers      1    36  11-Apr-63  House show   
1          Bruno Sammartino      1  2803  17-May-63  House show   
2               Ivan Koloff      1    21  18-Jan-71  House show   
3             Pedro Morales      1  1027   8-Feb-71  House show   
4              Stan Stasiak      1     9   1-Dec-73  House show   
5          Bruno Sammartino      2  1237  10-Dec-73  House show   
6    Superstar Billy Graham      1   296  30-Apr-77  House show   
8             Antonio Inoki      1     6  30-Nov-79  House show   
24                Bret Hart      1   174  12-Oct-92  House show   
30                   Diesel      1   358  26-Nov-94  House show   
223             Harley Race      1   183   1-Jan-75  House show   
224        Johnny Valentine      1    93   3-Jul-75  House show   
225              Terry Funk      1    18   9-Nov-75  House show   
226              Paul Jones      1   107  27-Nov-75  House show   
227      Blackjack Mulligan      1   217  13-Mar-76  House show   
228              Paul Jones      2    43  16-Oct-76  House show   
229      Blackjack Mulligan      2    11  28-Nov-76  House show   
230              Paul Jones      3     6   9-Dec-76  House show   
231      Blackjack Mulligan      2   204  15-Dec-76  House show   
232             Bobo Brazil      1    22   7-Jul-77  House show   
233               Ric Flair      1    84  29-Jul-77  House show   
234         Ricky Steamboat      1    72  21-Oct-77  House show   
235      Blackjack Mulligan      3    77   1-Jan-78  House show   
236           Mr. Wrestling      1    21  19-Mar-78  House show   
237               Ric Flair      2   253   9-Apr-78  House show   
238         Ricky Steamboat      2   105  17-Dec-78  House show   
239               Ric Flair      3   133   1-Apr-79  House show   
240             Jimmy Snuka      1   231   1-Sep-79  House show   
241               Ric Flair      4    98  19-Apr-80  House show   
242          Greg Valentine      1   121  26-Jul-80  House show   
243               Ric Flair      5    64  24-Nov-80  House show   
244             Roddy Piper      1   193  27-Jan-81  House show   
245          Wahoo McDaniel      1    31   8-Aug-81  House show   
246          Sgt. Slaughter      1   229   4-Oct-81  House show   
247          Wahoo McDaniel      2    17  21-May-82  House show   
248          Sgt. Slaughter      2    76   7-Jun-82  House show   
249          Wahoo McDaniel      3    74  22-Aug-82  House show   
250          Greg Valentine      1   163   4-Nov-82  House show   
251             Roddy Piper      2    14  16-Apr-83  House show   
252          Greg Valentine      2   228  30-Apr-83  House show   
253             Dick Slater      1   129  14-Dec-83  House show   
254         Ricky Steamboat      3    64  21-Apr-84  House show   
255          Wahoo McDaniel      4     7  24-Jun-84  House show   
256          Wahoo McDaniel      5   167   7-Oct-84  House show   
258         Tully Blanchard      1   130  21-Jul-85  House show   
260           Nikita Koloff      1   328  17-Aug-86  House show   
263           Barry Windham      1   283  13-May-88  House show   
266               Lex Luger      3   523  22-May-89  House show   
269                   Sting      1    86  25-Aug-91  House show   
310              Terry Funk      2     1  22-Sep-00  House show   
311             Lance Storm      2    36  23-Sep-00  House show   

                  location                           link  \
0         Washington, D.C.  /wiki/Buddy_Rogers_(wrestler)   
1             New York, NY         /wiki/Bruno_Sammartino   
2             New York, NY              /wiki/Ivan_Koloff   
3             New York, NY            /wiki/Pedro_Morales   
4         Philadelphia, PA             /wiki/Stan_Stasiak   
5             New York, NY         /wiki/Bruno_Sammartino   
6            Baltimore, MD   /wiki/Superstar_Billy_Graham   
8         Tokushima, Japan            /wiki/Antoni

Output the length of the resulting data frame (the subset of your data for which the conditional was true).

In [30]:
len(df_subset)

51

Output the first ten rows of your data set, the last ten rows of your data set, and a random ten rows of your data set (a sample of your data).

In [32]:
df.head(10)

name  reign  days       date               event  \
0            Buddy Rogers      1    36  11-Apr-63          House show   
1        Bruno Sammartino      1  2803  17-May-63          House show   
2             Ivan Koloff      1    21  18-Jan-71          House show   
3           Pedro Morales      1  1027   8-Feb-71          House show   
4            Stan Stasiak      1     9   1-Dec-73          House show   
5        Bruno Sammartino      2  1237  10-Dec-73          House show   
6  Superstar Billy Graham      1   296  30-Apr-77          House show   
7            Bob Backlund      1   648  20-Feb-78  WWF on MSG Network   
8           Antonio Inoki      1     6  30-Nov-79          House show   
9            Bob Backlund      2  1470  17-Dec-79  WWF on MSG Network   

           location                           link              belt  \
0  Washington, D.C.  /wiki/Buddy_Rogers_(wrestler)  WWE Championship   
1      New York, NY         /wiki/Bruno_Sammartino  WWE Championship   
2      New York, NY              /wiki/Ivan_Koloff  WWE Championship   
3      New York, NY            /wiki/Pedro_Morales  WWE Championship   
4  Philadelphia, PA             /wiki/Stan_Stasiak  WWE Championship   
5      New York, NY         /wiki/Bruno_Sammartino  WWE Championship   
6     Baltimore, MD   /wiki/Superstar_Billy_Graham  WWE Championship   
7      New York, NY             /wiki/Bob_Backlund  WWE Championship   
8  Tokushima, Japan            /wiki/Antonio_Inoki  WWE Championship   
9      New York, NY             /wiki/Bob_Backlund  WWE Championship   

  date_of_birth date_of_death                                   cause_of_death  
0     2/20/1921     6/26/1992                                              NaN  
1     10/6/1935     4/18/2018  Multiple organ failure due to heart problems[2]  
2     8/25/1942     2/18/2017                                     Liver cancer  
3    10/22/1942     2/12/2019                                              NaN  
4     4/13/1937     6/19/1997                                    Heart failure  
5     10/6/1935     4/18/2018  Multiple organ failure due to heart problems[2]  
6      6/7/1943           NaN                                              NaN  
7     8/14/1949           NaN                                              NaN  
8     2/20/1943           NaN                                              NaN  
9     8/14/1949           NaN                                              NaN

In [34]:
df.tail(10)

name  reign  days       date                          event  \
402         Carmella      1   131  10-Apr-18                      SmackDown   
403  Charlotte Flair      2    28  19-Aug-18                     SummerSlam   
404      Becky Lynch      2    91  16-Sep-18                 Hell in a Cell   
405            Asuka      1   100  16-Dec-18  TLC: Tables, Ladders & Chairs   
406  Charlotte Flair      3    13  26-Mar-19                      SmackDown   
407      Becky Lynch      3    41   8-Apr-19                WrestleMania 35   
408  Charlotte Flair      4     0  19-May-19              Money in the Bank   
409           Bayley      1   140  19-May-19              Money in the Bank   
410  Charlotte Flair      5     5   6-Oct-19                 Hell in a Cell   
411           Bayley      2   247  11-Oct-19                      SmackDown   

                location                       link  \
402      New Orleans, LA  /wiki/Carmella_(wrestler)   
403         Brooklyn, NY      /wiki/Charlotte_Flair   
404      San Antonio, TX          /wiki/Becky_Lynch   
405         San Jose, CA     /wiki/Asuka_(wrestler)   
406       Uncasville, CT      /wiki/Charlotte_Flair   
407  East Rutherford, NJ          /wiki/Becky_Lynch   
408         Hartford, CT      /wiki/Charlotte_Flair   
409         Hartford, CT    /wiki/Bayley_(wrestler)   
410       Sacramento, CA      /wiki/Charlotte_Flair   
411         Paradise, NV    /wiki/Bayley_(wrestler)   

                                   belt date_of_birth date_of_death  \
402  WWE SmackDown Women's Championship    10/23/1987           NaN   
403  WWE SmackDown Women's Championship      4/5/1986           NaN   
404  WWE SmackDown Women's Championship     1/30/1987           NaN   
405  WWE SmackDown Women's Championship     9/26/1981           NaN   
406  WWE SmackDown Women's Championship      4/5/1986           NaN   
407  WWE SmackDown Women's Championship     1/30/1987           NaN   
408  WWE SmackDown Women's Championship      4/5/1986           NaN   
409  WWE SmackDown Women's Championship     6/15/1989           NaN   
410  WWE SmackDown Women's Championship      4/5/1986           NaN   
411  WWE SmackDown Women's Championship     6/15/1989           NaN   

    cause_of_death  
402            NaN  
403            NaN  
404            NaN  
405            NaN  
406            NaN  
407            NaN  
408            NaN  
409            NaN  
410            NaN  
411            NaN

In [36]:
df.sample(10)

name  reign  days       date                  event  \
237     Ric Flair      2   253   9-Apr-78             House show   
216   Alexa Bliss      1   112  30-Apr-17                Payback   
314  Gen. Rection      2    49  26-Nov-00                 Mayhem   
88           Edge      3    21  23-Nov-08        Survivor Series   
264     Lex Luger      2    76  20-Feb-89        Chi-Town Rumble   
354  Jack Swagger      1    49  16-Jan-12                    Raw   
20     Hulk Hogan      4     1   3-Dec-91  This Tuesday in Texas   
228    Paul Jones      2    43  16-Oct-76             House show   
253   Dick Slater      1   129  14-Dec-83             House show   
52       Triple H      1    22  23-Aug-99                    Raw   

            location                         link  \
237    Charlotte, NC              /wiki/Ric_Flair   
216     San Jose, CA            /wiki/Alexa_Bliss   
314    Milwaukee, WI           /wiki/Gen._Rection   
88        Boston, MA        /wiki/Edge_(wrestler)   
264      Chicago, IL              /wiki/Lex_Luger   
354      Anaheim, CA           /wiki/Jack_Swagger   
20   San Antonio, TX             /wiki/Hulk_Hogan   
228   Greensboro, NC  /wiki/Paul_Jones_(wrestler)   
253       Shelby, NC            /wiki/Dick_Slater   
52          Ames, IA               /wiki/Triple_H   

                               belt date_of_birth date_of_death cause_of_death  
237  WWE United States Championship     2/25/1949           NaN            NaN  
216    WWE Raw Women's Championship      8/9/1991           NaN            NaN  
314  WWE United States Championship    11/10/1966           NaN            NaN  
88                 WWE Championship    10/30/1973           NaN            NaN  
264  WWE United States Championship      6/2/1958           NaN            NaN  
354  WWE United States Championship     3/24/1982           NaN            NaN  
20                 WWE Championship     8/11/1953           NaN            NaN  
228  WWE United States Championship     6/16/1942     4/18/2018            NaN  
253  WWE United States Championship     5/19/1951    10/18/2018            NaN  
52                 WWE Championship     7/27/1969           NaN            NaN

## Bonus

You may want to refer to the [Pandas Cheat Sheet](https://pandas.pydata.org/Pandas_Cheat_Sheet.pdf).

Sort your data frame by a particular column and output the result.

Pick a column in your data set with numeric data (integers or floats). Output the mean (average) and median of that column.